app/controllers/works_controller.rb

```diff
@@ -517,45 +517,52 @@ def destroy
 
   # POST /works/import
   def import
+    import_errors = ""
     # check to make sure we have some urls to work with
     @urls = params[:urls].split
 
-    if @urls.empty?
-      flash.now[:error] = ts('Did you want to enter a URL?')
-      render(:new_import) && return
+    if @urls.nil? || @urls.empty?
+      import_errors = ts("Did you want to enter a URL?")
     end
 
     # is external author information entered when import for others is not checked?
     if (params[:external_author_name].present? || params[:external_author_email].present?) && !params[:importing_for_others]
-      flash.now[:error] = ts('You have entered an external author name or e-mail address but did not select "Import for others." Please select the "Import for others" option or remove the external author information to continue.')
-      render(:new_import) && return
+      import_errors = ts("You have entered an external author name or e-mail address but did not select \"Import for others.\" " +
```

Comment:
I agree that returning just one error so you can surprise the user by returning a different one once they've fixed that isn't brilliant. It should also be a separate method which returns the error messages so the API can use it as well, instead of fattening this controller and duplicating error handling in the API, and there are error conditions which aren't covered at all.

However, as the purpose of the pull request was to expand the coverage of the WorksController, I wanted to minimise the scope of what I changed. (I couldn't stand all the `render view && returns` though 😄 ). Yanking the importing code out of this controller and improving it will happen in AO3-4315, so I will address it in that branch when we have the spoons to review it.